torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -435,17 +445,94 @@ def generate_table_visualization(self, feature_filter: str = "", module_fqn_filt
 
         print(table_str)
 
+    def _get_plotting_data(self, table_dict: Dict[str, Tuple[List, List]]) -> Tuple[bool, List, List]:
+        r"""
+        Takes in the table information as a dictionary and outputs both the x and y data
+        required to do the plotting.
+
+        Args:
+            table_dict ( Dict[str, Tuple[List, List]]): maps each of the:
+                TENSOR_LEVEL_INFO -> tensor_headers, tensor_table
+                CHANNEL_LEVEL_INFO -> channel_headers, channel_table
+
+        Returns a tuple of three elements:
+            If the data is per_channel or not
+            A list of the x values to plot
+            A list of the y values to plot
+        """
+        tensor_headers, tensor_table = table_dict[self.TABLE_TENSOR_KEY]
+        channel_headers, channel_table = table_dict[self.TABLE_CHANNEL_KEY]
+
+        # make sure it is only 1 feature that is being plotted
+        # get the number of features in each of these
+        tensor_info_features_count = len(tensor_headers) - ModelReportVisualizer.NUM_NON_FEATURE_TENSOR_HEADERS
+        channel_info_features_count = len(channel_headers) - ModelReportVisualizer.NUM_NON_FEATURE_CHANNEL_HEADERS
+
+        # keep track of per_channel or not
+        data_is_per_channel: bool = False
+
+        x_data: List = []
+        y_data: List[List] = []
+        # the feature will either be a tensor feature or channel feature
+        if tensor_info_features_count == 1:
+            # it should only be single value data, so easy to plot
+            for table_row_num, row in enumerate(tensor_table):
+                row_value = row[ModelReportVisualizer.NUM_NON_FEATURE_TENSOR_HEADERS]
+                if not type(row_value) == str:
```

Comment:
this also looks a bit hacky, is there a way to make this cleaner?

Reply:
yup added a couple more var declarations to make it more clear what we are using and why we are using it.

Reply:
cleaned it up to be half the size now